tests/test_sns.py

```diff
@@ -475,3 +475,113 @@ def test_sns_modify_add_and_remove_policy(self):
         self.assertTrue("AddMe" in statement_ids)
         self.assertTrue("RemoveMe" not in statement_ids)
         self.assertTrue("SpecificAllow" in statement_ids)
+
+    def test_sns_topic_encryption(self):
+        session_factory = self.replay_flight_data('test_sns_kms_related_filter_test')
+        kms = session_factory().client('kms')
+        p = self.load_policy(
+            {
+                'name': 'test-sns-kms-related-filter',
+                'resource': 'sns',
+                'filters': [
+                    {
+                        'TopicArn': 'arn:aws:sns:us-east-1:644160558196:test'
+                    },
+                    {
+                        'type': 'kms-key',
+                        'key': 'c7n:AliasName',
+                        'value': 'alias/skunk/trails'
+                    }
+                ]
+            },
+            session_factory=session_factory
+        )
+        resources = p.run()
+        self.assertTrue(len(resources), 1)
+        aliases = kms.list_aliases(KeyId=resources[0]['KmsMasterKeyId'])
+        self.assertEqual(aliases['Aliases'][0]['AliasName'], 'alias/skunk/trails')
+
+    def test_set_sns_topic_encryption(self):
+        session_factory = self.replay_flight_data('test_sns_set_encryption')
+        topic = 'arn:aws:sns:us-west-1:644160558196:test'
+
+        # default case, no key specified
+        p = self.load_policy(
+            {
+                'name': 'test-sns-kms-related-filter',
+                'resource': 'sns',
+                'filters': [
+                    {
+                        'TopicArn': topic
+                    },
+                    {
+                        'KmsMasterKeyId': 'absent'
+                    }
+                ],
+                'actions': [
+                    {
+                        'type': 'set-encryption'
+                    }
+                ]
+            },
+            session_factory=session_factory
+        )
+        resources = p.run()
+        self.assertEquals(len(resources), 1)
+
+        # disable encryption
```

Comment:
can you split these three scenarios out to separate tests.

Reply:
done